screens/LocationSelectScreen.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  ActivityIndicator,
+  Alert,
+} from "react-native";
+import { createStyleSheet, useStyles } from "react-native-unistyles";
+import SearchBar from "../components/header/SearchBar";
+import { Ionicons } from "@expo/vector-icons";
+import { useDispatch } from "react-redux";
+import {
+  getCurrentPositionAsync,
+  useForegroundPermissions,
+  PermissionStatus,
+} from "expo-location";
+import { getAddress } from "../services/location";
+import { useState, useEffect } from "react";
+import { setCity } from "../slices/citySlice";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { useNavigation, NavigationProp } from "@react-navigation/native";
+import { RootStackParamList } from "../types/navigation";
+
+const LocationSelectScreen = () => {
+  const { styles } = useStyles(stylesheet);
+  const dispatch = useDispatch();
+  const navigation: NavigationProp<RootStackParamList> = useNavigation();
+  const [locationPermissionInformation, requestPermission] =
+    useForegroundPermissions();
+  const [pickedLocation, setPickedLocation] = useState({ lat: 0, lng: 0 });
+  const [address, setAddress] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchAddress = async () => {
+      if (pickedLocation.lat && pickedLocation.lng) {
+        const fetchedAddress = await getAddress(
+          pickedLocation.lat,
+          pickedLocation.lng
+        );
+        setAddress(fetchedAddress);
+
+        dispatch(setCity(fetchedAddress));
+        try {
+          await AsyncStorage.setItem("selectedCity", fetchedAddress);
+        } catch (error) {
+          console.error(error);
+        }
```

Comment:
Nawet jeśli by nie używać tego `redux-persist`, fajnie by było gdyby robić to w jednym miejscu. Tak, żeby nigdy nie dało się dispatchować `setCity` bez zapisania do async storage. Można to wrzucić do samej akcji definiowanej w `createSlice`